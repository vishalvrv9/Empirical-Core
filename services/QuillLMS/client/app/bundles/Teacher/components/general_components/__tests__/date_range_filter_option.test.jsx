import React from 'react';
import { shallow } from 'enzyme';

import DateRangeFilterOption from '../date_range_filter_option';

describe('DateRangeFilterOption', () => {
  const mockFunction = jest.fn();

  const wrapper = shallow(
    <DateRangeFilterOption
      title='Example Title'
      onClickFunction={mockFunction}
    />
  );

  it('should render title from props', () => {
    expect(wrapper.text()).toMatch('Example Title');
  });

  it('should execute onClickFunction in props on click', () => {
    wrapper.simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
