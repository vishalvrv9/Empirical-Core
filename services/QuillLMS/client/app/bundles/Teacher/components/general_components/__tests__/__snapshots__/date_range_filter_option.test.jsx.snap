// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeFilterOption should match snapshot 1`] = `
<div
  className="calendar-prefill-option "
  onClick={[Function]}
>
  Example Title
</div>
`;
